src/utils/url.ts

```diff
@@ -26,7 +26,8 @@ export function parseConnectionString(connectionString: string): Options {
   }
 
   if (urlParts.pathname) {
-    options.database = urlParts.pathname.replace(/^\//, '');
+    // decode URI from urlParts.pathname value
+    options.database = decodeURI(urlParts.pathname.replace(/^\//, ''));
```

Comment:
Since we are not decoding the entire connection string, shouldn't we use `decodeURIComponent` here?

Reply:
Looking at[ the difference between the two](https://stackoverflow.com/questions/747641/what-is-the-difference-between-decodeuricomponent-and-decodeuri), it indeed seems better to use `decodeURIComponent`
Maybe we can add a unit test for decoding the special characters?

Reply:
I'd like to use the decodeURIComponent from start, but since we only want to decode the resulting encoding which are spaces and nonprintable characters I think using the decodeURIComponent is a bit overkill, as I'm not sure the database name will need the same thing as query strings.

Reply:
> Since we are not decoding the entire connection string, shouldn't we use `decodeURIComponent` here?

I'd like to use the decodeURIComponent from the start, but since we only want to decode the resulting encoding which are spaces and nonprintable characters I think using the decodeURIComponent is a bit overkill, as I'm not sure the database name will need the same things as query strings.

Reply:
> Looking at[ the difference between the two](https://stackoverflow.com/questions/747641/what-is-the-difference-between-decodeuricomponent-and-decodeuri), it indeed seems better to use `decodeURIComponent` Maybe we can add a unit test for decoding the special characters?

I'm not sure if the special characters is really allowed for database naming cause it may leading to unexpected errors.

Reply:
But yeah if you guys have good reason to use decodeURIComponent, yep I would agree to move from decodeURI

Reply:
Ah true, special characters in database naming might not be smart. We could test that with an integration test as well